src/basic/user-util.c

```diff
@@ -553,18 +553,18 @@ int take_etc_passwd_lock(const char *root) {
          * awfully racy, and thus we just won't do them. */
 
         if (root)
-                path = prefix_roota(root, "/etc/.pwd.lock");
+                path = prefix_roota(root, ETC_PASSWD_LOCK_PATH);
         else
-                path = "/etc/.pwd.lock";
+                path = ETC_PASSWD_LOCK_PATH;
 
         fd = open(path, O_WRONLY|O_CREAT|O_CLOEXEC|O_NOCTTY|O_NOFOLLOW, 0600);
         if (fd < 0)
-                return -errno;
+                return log_error_errno(errno, "Cannot open %s: %m", path);
```

Comment:
hmmpf, this turns take_etc_passwd_lock() into a function that logs on its own. is that really what we want? it's a "library" style function, hence I would not say no: the caller should log about this. In fact with this patch applied the callers will also log about this, hence you get duplicate logging which is very confusing.

Maybe instead of making this change, we should instead clean up the errors here, so that it is sufficiently clear to the caller which part is failing here, so that the caller can deal with this? For example we should turn EACCESS or EAGAIN on the fnctl() into EBUSY or so, to clean up the kernels borked handling error handling...

The other option is to rename this to "take_etc_password_lock_and_warn()" or so, but then all callers of this function should stop logging about this themselves, to avoid duplicate logging